app/views/projects/show.html.erb

```diff
@@ -0,0 +1,15 @@
+<h1><%= @project.name %></h1>
+
+<p>Material: <%= @project.material %></p><br>
+
+<p>Challenge Theme: <%= @challenge.theme %></p><br>
+
+<p>Number of Contestants: <%= @project.count_contestants %></p><br>
+
+<p>Average Contestant Experience: <%= @project.average_contestant_experience %> years</p><br>
+
+<%= form_with url: "/contestant_projects/#{@project.id}", method: :post, local: true do |form| %>
```

Comment:
This works and is set up fine; the only thing I would change is having a project_id in your url / path next to the "contestant_projects" resource. If I didn't know better or I didn't have access to your code, and I just saw `contestant_projects/12` as a url, I would assume that `12` refers to the ID of a specific `contestant_project` instance, and _not_ the project ID. 

If you want to send along additional data as part of this request, you can include a `form.hidden_field` in your form, and call it `:project_id` since you are correct that we need both the user-provided contestant ID, as well as the project that it will be associated with. Or, you can add it on as a query param - like `"/contestant_projects?project_id=#{@project.id}`" - either way, it will end up as part of your controller's `params`.